sockets/unix_socket.go

```diff
@@ -20,13 +89,18 @@ func NewUnixSocket(path string, gid int) (net.Listener, error) {
 	if err != nil {
 		return nil, err
 	}
-	if err := os.Chown(path, 0, gid); err != nil {
-		l.Close()
-		return nil, err
-	}
-	if err := os.Chmod(path, 0660); err != nil {
-		l.Close()
-		return nil, err
+
+	for _, op := range opts {
+		if err := op(path); err != nil {
+			l.Close()
+			return nil, err
+		}
 	}
+
 	return l, nil
 }
+
+// NewUnixSocket creates a unix socket with the specified path and group.
+func NewUnixSocket(path string, uid int) (net.Listener, error) {
```

Comment:
oh wait, this was the other way around :sweat_smile: 
`NewUnixSocket(path string, gid int)` and `os.Chown(path, 0, gid)`